core/common/src/main/java/alluxio/heartbeat/HeartbeatThread.java

```diff
@@ -38,27 +38,59 @@
   private HeartbeatTimer mTimer;
   private AlluxioConfiguration mConfiguration;
 
+  /**
+   * @param executorName the executor name defined in {@link HeartbeatContext}
+   * @param threadId the thread identifier
+   * @return the thread name combined of executorName and threadId, if threadId is empty or null,
+   *    return executorName
+   */
+  public static String generateThreadName(String executorName, String threadId) {
+    if (threadId == null || threadId.isEmpty()) {
+      return executorName;
+    }
+    return executorName + "-" + threadId;
+  }
+
   /**
    * Creates a {@link Runnable} to execute heartbeats for the given {@link HeartbeatExecutor}.
    *
    * This class is responsible for closing the given {@link HeartbeatExecutor} when it finishes.
    *
-   * @param threadName identifies the heartbeat thread name
+   * @param executorName identifies the heartbeat thread's executor, should be those defined in
+   *    {@link HeartbeatContext}
+   * @param threadId the thread identifier, normally, it is empty, but if a heartbeat
+   *    executor is started in multiple threads, this can be used to distinguish them, the heartbeat
+   *    thread's name is a combination of executorName and threadId
    * @param executor identifies the heartbeat thread executor; an instance of a class that
    *        implements the HeartbeatExecutor interface
    * @param intervalMs Sleep time between different heartbeat
    * @param conf Alluxio configuration
    */
-  public HeartbeatThread(String threadName, HeartbeatExecutor executor, long intervalMs,
```

Comment:
Each `HeartbeatExecutor` has a defined name in `HeartbeatContext`.
Originally, the assumption is that each `HeartbeatExecutor` will only run in one `HeartbeatThread`.
But it's not true for all heartbeat threads, such as the `ConfigHashSync` in `FileSystemContext`. 
So the new proposed `threadId` here is to discriminate between different heartbeat threads running the same `HeartbeatExecutor`, so that each `HeartbeatThread` has a unique name, which is the assumption in `HeartbeatScheduler`.

Reply:
got it, thx